### ISSUE TITLE
Improve UX

### ISSUE THREAD

**Issue by JosefJezek** [vaadin/vaadin-date-picker#343]
Could you add option attribute for default opened sidebar with year on mobile? Many users don't know click on year.

Touch target size in Material Design is 48px on touch screen and 40px on desktop. Please change size of days. Height 48px(40px) and width based on screen width.
https://material.io/guidelines/layout/metrics-keylines.html#metrics-keylines-touch-target-size

![image](https://cloud.githubusercontent.com/assets/2110319/23121949/a24f469e-f762-11e6-8fc4-f9e4da8bc43b.png)

Could you add arrow icons to sidebar with years?

![image](https://cloud.githubusercontent.com/assets/2110319/23121702/bd09f7be-f761-11e6-90df-7a8e2142abcf.png)

**Comment by JosefJezek** [comment]
@tomivirkki ?

**Comment by tomivirkki** [comment]
Thanks for the suggestion @JosefJezek. We’ll discuss these changes with the team and add them to the backlog if agreed upon.

**Comment by JosefJezek** [comment]
Please asap, thank you.

**Comment by limonte** [comment]
@jouni your opinion will be useful here

**Comment by jouni** [comment]
Right, I didn’t actually write a response here, even though I ended up doing some research, design and prototyping related to this.

That said, we currently have a long backlog of things to do before we have a chance to update the UX for the date-picker, unfortunately. I would like to, though, but our priorities are currently elsewhere.

Then for the actual discussion…

I actually started with a similar idea some years back when I was initially planning our new date picker (having a horizontal year scroller on small viewports / touch devices). But in the end I opted against that to keep the UX the same across all devices, and to keep the implementation easier.

I have been reconsidering the UI for touch devices lately, and here’s a mockup of one of the latest ones, done specifically for iOS:

![screen shot 2017-03-28 at 16 12 35](https://cloud.githubusercontent.com/assets/66382/24406694/6a090870-13d1-11e7-883e-8cb50738b911.png)

This is just ideation at this point, not a design we have settled on and will implement. It doesn’t take desktop use into account, for example, where horizontal scrolling is not a great way to navigate a long list.

I also made a quick functional prototype of that design, if you’re interested: http://jouni.me/date-picker-proto/demo/

**Comment by jouni** [comment]
> Touch target size in Material Design is 48px on touch screen and 40px on desktop. Please change size of days. Height 48px(40px) and width based on screen width with max-width 48px(40px).

That’s a valid point, and we will fix this in a future release.

**Comment by JosefJezek** [comment]
![image](https://cloud.githubusercontent.com/assets/2110319/25483665/16ff1df2-2b57-11e7-942b-66bebc985709.png)
